### DISCUSSION TITLE
[BP#75] Limiter le nombre de résultats (clause LIMIT)

### DISCUSSION THREAD

**Discussion by devcrafting** [cnumr/best-practices#221] [Bonnes pratiques] [answered]
Dans la série des BP sur SQL, celle-ce me semble trop restrictive/centrée sur un "détail".
J'aurais plutôt tendance à parler de limiter les données récupérées en base à ce qui est nécessaire pour le cas d'usage, comme discuté pour la BP 76 (cf. #102)

**Comment by ACTLEM** [comment]
Oui, sauf que là entre un LIMIT et un SELECT *, le SELECT * n'apporte pas grand chose (même si c'est d'après mes tests et curieux d'avoir d'autres retours) au contraire du LIMIT qui s'il est absent peut vraiment changer la donne. Personnellement, entre avoir "limiter les données récupérées" ou "utiliser la clause LIMIT", je préfère la deuxième car a une probabilité beaucoup plus forte d'être efficace.

**Comment by rlemaire** [comment]
Ça me parait être très très basique et évident pour les devs.
Il y a plein de moyen de réduire la quantité de données et la performance des requêtes (index par exemple) sur lesquels insister plus.

**Reply by ACTLEM** [reply]
Cela rejoint la discussion https://github.com/cnumr/best-practices/discussions/102 mais il y a une difficulté dans la rédaction de la BP, ce n'est pas évident de trouver une formulation opérationnelle et pas trop complexe sur les index.

**Comment by ACTLEM** [comment]
On ne touche pas à la BP (même si en général évident...quoique ^^)?

**Comment by ACTLEM** [comment, marked as answer]
Modification de la BP pour avoir une plus générale sur l'amélioration des requêtes SQL avec `LIMIT` comme exemple (voire aussi règle de validation): https://github.com/cnumr/best-practices/issues/315
